File: app/src/main/java/club/hutcwp/lifeutil/ui/reading/ReadFragment.java
package club.hutcwp.lifeutil.ui.reading;

import android.os.Bundle;
import android.support.design.widget.TabLayout;
import android.support.v4.app.Fragment;
import android.support.v4.content.ContextCompat;
import android.support.v4.view.ViewPager;

import java.util.List;

import club.hutcwp.lifeutil.R;
import club.hutcwp.lifeutil.adpter.ViewPagerAdapter;
import club.hutcwp.lifeutil.databinding.FragmentReadBinding;
import club.hutcwp.lifeutil.model.ReadCategory;
import club.hutcwp.lifeutil.ui.MainActivity;
import club.hutcwp.lifeutil.ui.base.BaseFragment;
import club.hutcwp.lifeutil.ui.base.IBaseView;
import hut.cwp.mvp.BindPresenter;

@BindPresenter(presenter = ReadPresenter.class)
public class ReadFragment extends BaseFragment<ReadPresenter, IBaseView> implements IBaseView<ReadCategory> {

    private FragmentReadBinding binding;

    @Override
    protected int getLayoutId() {
        return R.layout.fragment_read;
    }

    @Override
    protected void initViews() {
        binding = (FragmentReadBinding) getBinding();
        binding.toolbar.setTitle("阅读");
        ((MainActivity) getActivity()).initDrawer(binding.toolbar);
    }

    protected void lazyFetchData() {
        getPresenter().getCategory();
    }

    /**
     * 初始化TabLayout
     *
     * @param readCategories 标签类
     */
    @Override
    public void initTabLayout(List<ReadCategory> readCategories) {
        setUpViewPager(binding.viewPager, readCategories);
        binding.viewPager.setOffscreenPageLimit(binding.viewPager.getAdapter().getCount());
        binding.tablayout.setSelectedTabIndicatorColor(ContextCompat.getColor(getActivity(), R.color.white));
        binding.tablayout.setupWithViewPager(binding.viewPager);
        binding.tablayout.setTabMode(TabLayout.MODE_SCROLLABLE);
    }

    @Override
    public void setUpViewPager(ViewPager viewPager, List<ReadCategory> readCategories) {
        ViewPagerAdapter adapter = new ViewPagerAdapter(getChildFragmentManager());
        for (ReadCategory category : readCategories) {
            Fragment fragment = new ReadCategoryFragment();
            Bundle data = new Bundle();
            data.putString("url", category.getUrl());
            fragment.setArguments(data);
            adapter.addFrag(fragment, category.getName());
        }
        viewPager.setAdapter(adapter);
    }
}

File: app/src/main/java/club/hutcwp/lifeutil/ui/home/top/ReadFragment.kt
package club.hutcwp.lifeutil.ui.home.top

import android.os.Bundle
import android.util.Log
import com.google.android.material.tabs.TabLayout
import androidx.core.content.ContextCompat
import androidx.viewpager.widget.ViewPager

import club.hutcwp.lifeutil.R
import club.hutcwp.lifeutil.adpter.ViewPagerAdapter
import club.hutcwp.lifeutil.databinding.FragmentReadBinding
import club.hutcwp.lifeutil.entitys.ReadCategory
import club.hutcwp.lifeutil.ui.MainActivity
import club.hutcwp.lifeutil.ui.base.BaseFragment
import club.hutcwp.lifeutil.ui.home.sub.news.NewsFragment
import hut.cwp.mvp.BindPresenter

@BindPresenter(presenter = ReadPresenter::class)
class ReadFragment : BaseFragment<ReadPresenter, ReadFragment>(), IHome<ReadCategory> {

    private var fragmentReadBinding: FragmentReadBinding? = null

    override fun getLayoutId(): Int {
        return R.layout.fragment_read
    }

    override fun initViews() {
        fragmentReadBinding = binding as FragmentReadBinding
        fragmentReadBinding!!.toolbar.title = "阅读"
        (activity as MainActivity).initDrawer(fragmentReadBinding!!.toolbar)
    }

    override fun lazyFetchData() {
        presenter.getCategory()
    }

    /**
     * 初始化TabLayout
     *
     * @param categories 标签类
     */
    override fun initTabLayout(categories: List<ReadCategory>) {
        Log.i("cwp","initTabLayout")
        setUpViewPager(fragmentReadBinding!!.viewPager, categories)
        fragmentReadBinding!!.viewPager.offscreenPageLimit = fragmentReadBinding!!.viewPager.adapter!!.count
        fragmentReadBinding!!.tablayout.setSelectedTabIndicatorColor(ContextCompat.getColor(activity!!, R.color.white))
        fragmentReadBinding!!.tablayout.setupWithViewPager(fragmentReadBinding!!.viewPager)
        fragmentReadBinding!!.tablayout.tabMode = TabLayout.MODE_SCROLLABLE
    }

    override fun setUpViewPager(viewPager: ViewPager, readCategories: List<ReadCategory>) {
        val adapter = ViewPagerAdapter(childFragmentManager)
        for (category in readCategories) {
            val fragment = NewsFragment()
            val data = Bundle()
            data.putString("url", category.url)
            fragment.arguments = data
            adapter.addFrag(fragment, category.name!!)
        }
        viewPager.adapter = adapter
    }
}
